src/index.js

```diff
@@ -53,45 +54,56 @@ function createBabelPluginFactory(customCallback = returnObject) {
 			externalHelpersWhitelist,
 			include,
 			runtimeHelpers,
+			transformGenerated,
 			...babelOptions
 		} = unpackOptions(pluginOptions);
 
+		if (transformGenerated) {
+			return {
+				name: 'babel-output',
+
+				renderStart() {
+					if (pluginOptions.include || pluginOptions.exclude || pluginOptions.extensions) {
+						warnOnce(
+							this,
+							'The "include", "exclude" and "extensions" options are ignored when using the "transformGenerated" option.',
+						);
+					}
+				},
+
+				renderChunk(code, chunk, options) {
+					return transformCode(
+						code,
+						{ ...babelOptions, filename: options.file || (options.dir && `${options.dir}/placeholder.js`) },
```

Comment:
In case of the dir option i would use some more distinct filename for the placeholder, smth like `__rollup_babel_placeholder__.js` or similar

Also - couldnt we provide a final filename for entry files?

Reply:
> Also - couldnt we provide a final filename for entry files

So you would want to use the entry file as base for where to look for configuration? This does not seem ideal to me as a bundle could contain files from many different places and this could interfere with the situation where you apply the plugin both on the input and the output. Having the output files as base for where we are looking for configuration would allow keeping those two separate.

> In case of the dir option i would use some more distinct filename for the placeholder, smth like __rollup_babel_placeholder__.js

Why, as I understand, this name does not serve any purpose at all except to tell Babel where to start searching for configs. Maybe I will just change it to `_.js` to make clear the name does not matter.

Reply:
I'd lean toward making this option be `null` by default as a opt-in for users. This means we'd expect users to use a `babel.config.js`. It's only `.babelrc` config files that are loaded per-filename.

Reply:
Ah, I was not aware about `babel.config.js` files (and also updated the Readme wrongly). Yes, that makes sense. So you suggest we do not pass in the `filename` in any case but allow users to pass in a filename?

Reply:
Great. In all honesty, it may be eve better to not expose the option at all. Perhaps we could see if anyone requests it, to understand why they want it? Most of the time you'd want a filename to
a) Load a specific config via `.babelrc`
b) To allow `ignore`/`only`/`test`/`include`/`exclude` to work

but in the context of a whole-bundle compilation those don't conceptually apply.

Reply:
Makes sense. I removed the "dir"/"filename" handling and changed the docs to mention that we only respect project-wide configuration based on the current working directory. It would still be possible to pass in a `filename`, but it is not documented, and I do not see any danger in leaving this open.